front/src/utils/index.ts

```diff
@@ -10,6 +10,36 @@ export function formatToHumanReadableDate(date: Date | string) {
   }).format(parsedJSDate);
 }
 
-export const getLogoUrlFromDomainName = (domainName?: string): string => {
-  return `https://api.faviconkit.com/${domainName}/144`;
-};
+export function sanitizeURL(link: string | null | undefined) {
+  return link
+    ? link.replace(/(https?:\/\/)|(www\.)/g, '').replace(/\/$/, '')
+    : '';
+}
+
+export function getLogoUrlFromDomainName(
+  domainName?: string,
+): string | undefined {
+  const sanitizedDomain = sanitizeURL(domainName);
+
+  if (!sanitizedDomain) return;
+
+  const prioritizeWWW = ['apple.com'].includes(sanitizedDomain);
+
+  const prefixes = prioritizeWWW
+    ? ['www.', '', 'https://']
+    : ['', 'www.', 'https://'];
+
+  for (const prefix of prefixes) {
+    const img = document.createElement('img');
```

Comment:
It seems weird to create a fake image here. Let's use fetch api instead if possible